native/src/main/java/io/ballerina/lib/data/yaml/utils/OptionsUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.data.yaml.utils;
+
+import io.ballerina.lib.data.yaml.common.Types;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+/**
+ * This class will parse the user configs to Java records.
+ *
+ * @since 0.1.0
+ */
+public class OptionsUtils {
+
+    private OptionsUtils() {
+    }
+
+    public record WriteConfig(int indentationPolicy, int blockLevel, boolean canonical, boolean useSingleQuotes,
+                              boolean forceQuotes, Types.YAMLSchema schema, boolean isStream, boolean flowStyle) {
+    }
+
+    public static WriteConfig resolveWriteOptions(BMap<BString, Object> options) {
+        Object indentationPolicy = options.get(Constants.INDENTATION_POLICY);
+        Object blockLevel = options.get(Constants.BLOCK_LEVEL);
+        Object canonical = options.get(Constants.CANONICAL);
+        Object useSingleQuotes = options.get(Constants.USE_SINGLE_QUOTES);
+        Object forceQuotes = options.get(Constants.FORCE_QUOTES);
+        Object schema = options.get(Constants.SCHEMA);
+        Object isStream = options.get(Constants.IS_STREAM);
+        Object flowStyle = options.get(Constants.FLOW_STYLE);
+
+        return new WriteConfig(Math.toIntExact(((Long) indentationPolicy)), Math.toIntExact(((Long) blockLevel)),
+                ((Boolean) canonical), ((Boolean) useSingleQuotes), ((Boolean) forceQuotes),
+                Types.YAMLSchema.valueOf(((BString) schema).getValue()), ((Boolean) isStream), ((Boolean) flowStyle));
+    }
+
+    public record ReadConfig(Types.YAMLSchema schema, boolean allowAnchorRedefinition,
+                             boolean allowMapEntryRedefinition, boolean allowDataProjection,
+                             boolean nilAsOptionalField, boolean absentAsNilableType, boolean strictTupleOrder) {
+    }
+
+    public static ReadConfig resolveReadConfig(BMap<BString, Object> options) {
+        Object schema = options.get(Constants.SCHEMA);
+        Object allowAnchorRedefinition = options.get(Constants.ALLOW_ANCHOR_REDEFINITION);
+        Object allowMapEntryRedefinition = options.get(Constants.ALLOW_MAP_ENTRY_REDEFINITION);
+        Object allowDataProjection = options.get(Constants.ALLOW_DATA_PROJECTION);
+        Object isStream = options.get(Constants.IS_STREAM);
```

Comment:
Instead of intialize them as `Object`, better to initilize as actual type such as `boolean`, `BString`